tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -101,39 +101,76 @@ def __init__(self, manager):
         self.manager = manager
 
     def get_resources(self, query):
-        session = local_session(self.manager.session_factory)
+        session = self.manager.session_factory(
+            quota_max_calls=300,
+            quota_period=60,
+            use_rate_limiter=True)
+
         if query is None:
             query = {}
         if 'scope' not in query:
             query['scope'] = 'projects/%s' % session.get_default_project()
         if 'assetTypes' not in query:
             query['assetTypes'] = [self.manager.resource_type.asset_type]
 
-        search_client = session.client('cloudasset', 'v1p1beta1', 'resources')
-        resource_client = session.client('cloudasset', 'v1', 'v1')
+        asset_client = session.client('cloudasset', 'v1', 'v1')
         resources = []
-
-        results = list(search_client.execute_paged_query('searchAll', query))
-        for resource_set in chunks(itertools.chain(*[rs['results'] for rs in results]), 100):
+        query['pageSize'] = 500
+        results = list(asset_client.execute_paged_query(
+            'searchAllResources', query))
+
+        if not results or not results[0]:
```

Comment:
what is results[0] checking for. is this checking if the first response is None?

Reply:
a paged query returns an empty dict  on no results, so `results` ends up looking like `[{}]`

Reply:
so would we need the `not results` check then if results will always return at least a list with an empty dict?

Reply:
yeah, but its always good practice re defensive to check the list has contents before direct access to an element